src/components/Task/TaskList/TaskList.tsx

```diff
@@ -0,0 +1,91 @@
+import "./TaskList.css";
+
+import { useState, useRef } from "react";
+import { CheckListCircle } from "../../../assets/checkListCircle";
+import EditIcon from "../../../assets/Edit-Icon";
+import DeleteButton from "../../../assets/Delete-Icon";
+
+interface TaskListProps {
+  index: number;
+  task: { id: number; name: string; completed: boolean };
+  onToggleTask: (index: number) => void;
+  onDeleteTask: (index: number) => void;
+  onEditTask: (index: number, newName: string) => void;
+}
+
+function TaskList({
+  index,
+  task,
+  onToggleTask,
+  onDeleteTask,
+  onEditTask,
+}: TaskListProps) {
+  const [isEditing, setIsEditing] = useState(false);
+  const taskNameRef = useRef<HTMLParagraphElement>(null);
+
+  const handleEdit = () => {
+    console.log("Edit button clicked");
+    setIsEditing(true);
+    setTimeout(() => {
+      taskNameRef.current?.focus();
```

Comment:
what is the need of adding 0 sec delay?

Reply:
Originally added more delay because on edit click the task box should be in focus. I agree totally that a 0 second delay has no purpose . I will remove it